base/context-overlay.js

```diff
@@ -0,0 +1,148 @@
+
+    this.overlayArrow.addEventListener('animationend', () => {
+      this.overlayArrow.classList.remove('fade-in')
+    })
+
+    this.options = null
+    this.position = null
+
+    this._debRender = debounce(this._render, 200).bind(this)
+  }
+
+  show (options) {
+    document.body.appendChild(this.overlayEl)
+
+    this.options = options
+    this.overlayEl.classList.add('show', ...options.classNames || [])
+    this._render()
+
+    window.addEventListener('resize', this._debRender)
+    window.addEventListener('scroll', this._debRender)
+  }
+
+  hide () {
+    document.body.removeChild(this.overlayEl)
+    this.overlayInnerEl.innerHTML = ''
+    this.overlayEl.style.cssText = ''
+    this.overlayInnerEl.style.cssText = ''
+    this.overlayEl.classList.remove('show')
+
+    window.removeEventListener('resize', this._debRender)
+    window.removeEventListener('scroll', this._debRender)
+  }
+
+  getPosition () { return this.position }
+
+  _render () {
+    if (!this.overlayEl.classList.contains('show')) return
+
+    this.overlayEl.classList.toggle('showArrow', this.options.showArrow)
+
+    let {
+      msg,
+      targetElement,
+      targetRect,
+      outerRect = document.body.getBoundingClientRect(),
+      offset,
+      pointerCoords,
+      verticalAlign = 'bottom'
+    } = this.options
+
+    let {
+      left: x,
+      top: y,
+      width,
+      height
+    } = targetRect || targetElement.getBoundingClientRect()
+
+    if (y + height > window.innerHeight / 2) {
+      verticalAlign = 'top'
+    }
+
+    let msgHtmlNode = toHtml(msg)
+
+    const arrowHeight = this.options.showArrow ? 10 : 0
+
+    if (offset) {
+      x += offset.x || 0
+      y += offset.y || 0
+      width += offset.width || 0
+      height += offset.height || 0
+    }
+
+    let ttLeft = x + width / 2
+
+    // if the element is in the lower half of the screen than align the overlay to the top side
+    let ttTop = y + (verticalAlign === 'bottom' ? height + arrowHeight : -arrowHeight)
+
+    this.overlayArrow.classList.add('fade-in')
+
+    this.overlayEl.classList.toggle('arrowTop', verticalAlign === 'bottom')
+    this.overlayEl.classList.toggle('arrowBottom', verticalAlign === 'top')
+
+    if (pointerCoords) {
+      // centering on the mouse pointer horizontally
+      ttLeft = x + pointerCoords.x
+    }
+
+    const oldWidth = this.overlayInnerEl.style.width
+    this.overlayInnerEl.style.width = 'auto'
+
+    this.overlayInnerEl.innerHTML = ''
+    this.overlayInnerEl.appendChild(msgHtmlNode)
+
+    this.overlayEl.style.cssText = `left:${ttLeft}px; top:${ttTop}px;`
+    // calculating the actual overlay width
+    const ttWidth = msgHtmlNode.offsetWidth
+    const ttHeight = msgHtmlNode.offsetHeight
+
+    this.overlayInnerEl.style.width = oldWidth
+
+    const justToForceRedraw = this.overlayInnerEl.offsetWidth
+    // the following line is to avoid the "no-unused-vars" error
+    if (justToForceRedraw === 0) console.log(justToForceRedraw)
```

Comment:
Should be able to just add `// eslint-disable-line no-unused-vars` to the `justToForceRedraw` line, standard listens to eslint comments

Reply:
👍